### PR TITLE
Added threadId to Element

### PR DESCRIPTION
**Pull request by mayaankvad**
Adding this as prep for a custom data layer. This change will allow more flexible custom data layers, especially document ones like Dynamo.

Currently you can get the threadId in create element:
```python
async def create_element(self, element: "Element"):
        thread_id = element.thread_id)
```

This change adds the thread_id param to delete element:
```python
async def delete_element(self, element_id: str, thread_id: Optional[str]):
```

All current data layers have been modified to accept the new param. The [Custom Data Layer API Docs](https://docs.chainlit.io/api-reference/data-persistence/custom-data-layer) will need to be updated. Should not be a breaking change but should be a mention in a migration guide.

### CODE REVIEW

File: cypress/e2e/data_layer/main.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ async def get_thread_author(self, thread_id: str):
         return "admin"
 
     async def list_threads(
-        self, pagination: cl_data.Pagination, filter: cl_data.ThreadFilter
+        self, pagination: cl_data.Pagination, filters: cl_data.ThreadFilter
```
Comment (mayaankvad) [inline cypress/e2e/data_layer/main.py:117]: Fixed error:

```
Parameter 'filters' has been renamed to 'filter' in overriding 'TestDataLayer.list_threads' method PylintW0237:arguments-renamed

Method "list_threads" overrides class "BaseDataLayer" in an incompatible manner
  Parameter 3 name mismatch: base parameter is named "filters", override parameter is named "filter" Pylance reportIncompatibleMethodOverride
__init__.py(118, 15): Overridden method
```

File: backend/chainlit/data/sql_alchemy.py
Comment (mayaankvad) [inline backend/chainlit/data/sql_alchemy.py:1]: Accidentally auto formatted this to change more lines than intended. But the changes are all just putting the logger statement on a new line. Leaving this minor style fix in, happy to revert if needed.

File: backend/chainlit/data/__init__.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ async def get_element(
         pass
 
     @queue_until_user_message()
-    async def delete_element(self, element_id: str):
+    async def delete_element(self, thread_id: str, element_id: str):
```
Comment (willydouhard) [inline backend/chainlit/data/__init__.py:97]: this is a breaking change, i would put it after element_id as optional?

File: backend/chainlit/data/sql_alchemy.py
Comment (mayaankvad) [inline backend/chainlit/data/sql_alchemy.py:1]: git hook `pnpm lint-staged` has made changes everywhere here to add the logger on a new line.

**Pull request merged**